### DISCUSSION TITLE
avatar not display correct in latest image 0.7.5-rc2

### DISCUSSION THREAD

**Discussion by elvis-cai** [danny-avila/LibreChat#4211] [Troubleshooting]
### What happened?

<img width="593" alt="image" src="https://github.com/user-attachments/assets/7c0613d0-a480-4e8d-b7ac-b91fb0e3ad13">
the image is not displayed correctly after login with okta.
Let me know if I am missing anything, thx.

### Steps to Reproduce

deploy with image: ghcr.io/danny-avila/librechat:v0.7.5-rc2, login with sso and check user avatar.

### Screenshots

<img width="1245" alt="image" src="https://github.com/user-attachments/assets/6d9ff5eb-4af8-41d7-9407-77ebc441a549">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
likely your OIDC provider doesn't return an image or rather it doesn't return an avatar in the expected userinfo field (currently only `userinfo.picture`, not unique to Entra but not universally implemented across all OIDC providers)

**Reply by elvis-cai** [reply]
Hi @danny-avila it works with previous old version, we are running `v0.7.4` in the past 
here's the working example 
<img width="546" alt="image" src="https://github.com/user-attachments/assets/ec63cef6-53c9-48bf-bc65-f42589fd4bd3">

**Reply by danny-avila** [reply]
This is not an avatar but rather the lack of one that generates the initials based on the username on the client side.

As I wrote in the other comment, it would be helpful to see this particular log from debug logs, with any sensitive info redacted: `[openidStrategy] very login tokenset and userinfo`. This would confirm what is happening exactly on your end. 

If you are using docker, they are saved in the `./logs` directory at project root.

Have a user login and then check the latest debug logs (files starting with "debug" in name, i.e. `logs/debug-2024-09-24.log`)

**Reply by danny-avila** [reply]
It might also be the recently added "alt" that is causing this. I can investigate that, too.

**Reply by elvis-cai** [reply]
here's the debug log when start a new conversation https://gist.github.com/elvis-cai/fadcc8d6b8b6100d5753948ee743a7f1
here's the log during okta login 
```

2024-09-24T23:49:46.224Z info: [openidStrategy] verify login openidId: 00u2co54huvbbqRsw1d7
2024-09-24T23:49:46.224Z debug: [openidStrategy] very login tokenset and userinfo
{
    tokenset.token_type: "Bearer",
    tokenset.expires_at: 1727225385,
    tokenset.access_token: "eyJra........ [truncated]",
    tokenset.scope: "openid profile email",
    tokenset.id_token: "eyJraWQiOi....... [truncated]",
    userinfo.sub: "00u2co54huvbbqRsw1d7",
    userinfo.name: "Elvis Cai",
    userinfo.locale: "AU",
    userinfo.email: "elvis.cai@xxx.com",
    userinfo.nickname: "Elvis",
    userinfo.preferred_username: "elvis.cai@xxx.com",
    userinfo.given_name: "Elvis",
    userinfo.middle_name: "",
    userinfo.family_name: "Cai",
    userinfo.zoneinfo: "Australia/Melbourne",
    userinfo.updated_at: 1710213694,
    userinfo.email_verified: true,
}
2024-09-24T23:49:46.227Z info: [openidStrategy] user found with openidId: 00u2co54huvbbqRsw1d7
2024-09-24T23:49:46.243Z info: [openidStrategy] login success openidId: 00u2co54huvbbqRsw1d7 | email: elvis.cai@xxx.com | username: elvis


```

**Reply by elvis-cai** [reply]
find this [commit](https://github.com/danny-avila/LibreChat/commit/98b437edd504245741d1abfebfda5f2d44b9ffa7#diff-ffb2b114a810eed33265949ff3c490e9934dfa179586c33af6da3ca0463fc77dL42)  maybe related

**Reply by danny-avila** [reply]
I don’t think it’s frontend related because images from my Entra login are showing up. I will setup okta and see if I can reproduce it with that

**Reply by danny-avila** [reply]
Just setup Okta, my signin looks similar

```bash
{
    tokenset.token_type: "Bearer",
    tokenset.expires_at: 1727274196,
    tokenset.access_token: "redacted",
    tokenset.scope: "email profile openid",
    tokenset.id_token: "redacted",
    userinfo.sub: "00uj8ywu8gytVmMEO697",
    userinfo.name: "D A",
    userinfo.locale: "en_US",
    userinfo.email: "redacted@pm.me",
    userinfo.preferred_username: "redacted@pm.me",
    userinfo.given_name: "D",
    userinfo.family_name: "A",
    userinfo.zoneinfo: "America/Los_Angeles",
    userinfo.updated_at: 1727268773,
    userinfo.email_verified: true,
}
```

![image](https://github.com/user-attachments/assets/32237ce9-cd1b-4615-8244-7ccec1dd1ec7)
![image](https://github.com/user-attachments/assets/4e4f4359-5de8-4320-bd14-622be7dae15d)


In my case, the frontend dependency that is creating the SVG is working as intended, because profile picture is not present from Okta (as should be in your case)


--- 

@elvis-cai In your HTML inspect screenshot, it shows no `src` so likely the dicebear dependency is failing or never getting activated. I'm not what would cause that, but I've tested on 3 different browsers and it's working as intended.

**Reply by elvis-cai** [reply]
Thanks @danny-avila for reproducing the issue, interesting, I found my profile image is not rendered correctly either, it works perfectly in 0.7.4 
<img width="663" alt="image" src="https://github.com/user-attachments/assets/ac34d5d5-472e-4953-809b-a0a2f4083607">

**Reply by elvis-cai** [reply]
<img width="1329" alt="image" src="https://github.com/user-attachments/assets/e5619892-f02f-45fa-a644-232e7d840ecd">
0.7.4

**Reply by danny-avila** [reply]
do you mind sharing the user object of your account? you can get this from the network tab in dev tools, filter by "user" and refresh:


![image](https://github.com/user-attachments/assets/bbdd6509-b6e9-432c-b0d2-05d691e87847)

**Reply by elvis-cai** [reply]
here you go @danny-avila  :) 
<img width="1398" alt="image" src="https://github.com/user-attachments/assets/5bff1ec3-e068-4db1-8fd1-77b132a88e3b">
<img width="1356" alt="image" src="https://github.com/user-attachments/assets/1938f001-a140-43f7-a017-837fcd74a1d2">

**Reply by elvis-cai** [reply]
thanks @sn-smaier 👍  , as you can see from the screenshot `avato`r is `""` in my case , cc @danny-avila is there anything we could do to fix this issue? 🙏

**Comment by drewdresser** [comment]
I'm hitting this with Okta too. Did something related to avatar display or is it a change on the Okta side?

**Reply by danny-avila** [reply]
File history: https://github.com/danny-avila/LibreChat/commits/3e0f95458f1e1085b1268b7dea2c602d8f902c56/api/strategies/openidStrategy.js

when was your last update?

**Reply by danny-avila** [reply]
Would be helpful to see this particular log from debug logs, with any sensitive info redacted: `[openidStrategy] very login tokenset and userinfo`

**Comment by sn-smaier** [comment]
It's caused by the change from `user?.avatar || avatarSrc` to `user?.avatar ?? avatarSrc`
the operator causes this issue if user.avatar in the database is saved as empty string instead of null

**Reply by danny-avila** [reply]
Thanks for the tip, I'll fix this today

**Comment by danny-avila** [comment]
Please let me know if this PR fully addresses the issue when merged: https://github.com/danny-avila/LibreChat/pull/4321

**Reply by elvis-cai** [reply]
thanks @danny-avila , can confirm it works perfectly for me now 👍
